proposal/WarClans612/proposal.rst

```diff
@@ -0,0 +1,154 @@
+library that is able to do calculation with those 2 storage method. The graph class will then
+inherit from the matrix class to be used as further extension. However, in this project, we will
+focus more of feature implementation for the matrix rather than the graph.
+
+Perspective users
+=================
+
+- Research focus operation related to sparse matrix
+- Data analytic team that operates with huge nodes in graph
+- Company with huge amount of graph relation data collection that would like to decrease its storage usage
+
+System architecture
+===================
+
+Input:
+
+- Graph can be initialized as an empty graph
+- Provided method to add nodes into the graph
+- Provided method to add edges into the graph
+
+Output:
+
+- Provided method to store the matrix into a text file
+- Provided method to manipulate the matrix
+
+Modularization
+-----------------
+
+- SparseMatrix Class
+    - SparseMatrix Manipulation
+    - Storage
+    - Computation
+- Graph Class (inherit from Matrix Class):
+    - Graph Manipulation
+    - Storage (similar to matrix, graph can be represented as matrix)
+- Testing Module
+- Config Module
+- Example Module
+
+API description
+===============
+
+- SparseMatrix Class
+    - ``SparseMatrix()``
+    - ``SparseMatrix(vector&)``
+    - ``SparseMatrix(vector<vector>&)``
+    - ``SparseMatrix(SparseMatrix&)``
+    - ``void loadSparseMatrix(string)``
+    - ``void saveSparseMatrix(string)``
+    - ``void resetSparseMatrix()``
+    - 
+    - ``const float&  operator()(int, int)``
+    - ``void          operator()(int, int, float)``
+    - ``SparseMatrix& operator=(const SparseMatrix&)``
+    - ``SparseMatrix  operator+(const SparseMatrix&)``
+    - ``SparseMatrix  operator-(const SparseMatrix&)``
+    - ``SparseMatrix  operator*(const float&)``
+    - ``SparseMatrix  operator*(const SparseMatrix&)``
+    - ``SparseMatrix  operator*(const SparseMatrix&)``
+    - ``SparseMatrix  operator/(const SparseMatrix&)``
+    - 
+    - ``addNode(int, float)``
+    - ``removeNode(int)``
+    - ``addEdge(int, float)``
+    - ``removeEdge(int)``
+    - 
+    - ``findIndex(float)``
+    - ``int dim(float)``
+
+- Graph Class (currently Wrapper for SparseMatrix Class)
+
+Engineering infrastructure
+==========================
+
+SparseMatrix class will be implemented first with Graph Class inheriting its module.
+Initialization, loading, and saving module will be the most critical parts at the beggining.
+The above modules will enable us to verify the correctness of CSC and CSR implementation
+by comparing it with python Scipy Package (used as testing purposes).
+
+The next part is implementing its computation operator, which also will be verified with
+Scipy Package.
+
+PyBind11 will be used to glue C++ module and Python called to be tested.
+Pytest will then be used to implement unittest capability.
+Git will be used to track and maintain version.
+
+Schedule
+========
+
+- 11/02 - 11/08 -> Implementing SparseMatrix module shell
```

Comment:
This project looks promising to impact.  I think it may be beneficial for you to consider longer-term development.